components/eamxx/src/control/atmosphere_surface_coupling_exporter.cpp

```diff
@@ -196,6 +204,40 @@ void SurfaceCouplingExporter::initialize_impl (const RunType /* run_type */)
   // Copy data to device for use in do_export()
   Kokkos::deep_copy(m_column_info_d, m_column_info_h);
 
+  // Set the number of exports from eamxx or set to a constant, default type = EAMXX
+  using vos_type = std::vector<std::string>;
+  using vor_type = std::vector<Real>;
+  m_export_source = view_1d<DefaultDevice,ExportType>("",m_num_scream_exports);
+  Kokkos::deep_copy(m_export_source,EAMXX);  // The default is that all export variables will be derived from the EAMxx state.
+  m_num_eamxx_exports = m_num_scream_exports;
+ 
+  if (m_params.isSublist("prescribed_constants")) {
+    auto export_constant_params = m_params.sublist("prescribed_constants");
+    EKAT_REQUIRE_MSG(export_constant_params.isParameter("fields"),"Error! surface_coupling_exporter::init - prescribed_constants does not have 'fields' parameter.");
+    EKAT_REQUIRE_MSG(export_constant_params.isParameter("values"),"Error! surface_coupling_exporter::init - prescribed_constants does not have 'values' parameter.");
+    auto export_constant_fields = export_constant_params.get<vos_type>("fields");
+    auto export_constant_values = export_constant_params.get<vor_type>("values");
+    EKAT_REQUIRE_MSG(export_constant_fields.size()==export_constant_values.size(),"Error! surface_coupling_exporter::init - prescribed_constants 'fields' and 'values' are not the same size");
+    if (export_constant_fields.size()>0) {
+      // Determine which fields need constants
+      m_export_constants = view_1d<DefaultDevice,Real>("",m_num_scream_exports);
+      for (int i=0; i<m_num_scream_exports; ++i) {  // TODO: This loop would probably be simpler if we just checked which "i" corresponded to each name in the fields list.
+        std::string fname = m_export_field_names[i];
+        auto loc = std::find(export_constant_fields.begin(),export_constant_fields.end(),fname);
+        if (loc != export_constant_fields.end()) {
+          const auto pos = loc-export_constant_fields.begin();
+          m_export_source(i) = CONSTANT;
```

Comment:
This one and the one 3 lines below are likely the reasons of the fail on weaver. You're accessing dev views from host. You need a host mirror, and then deep copy.